RELEASING.md

```diff
@@ -0,0 +1,51 @@
+# Releasing OpenTelemetry Rust
+
+The primary audience for this is the SIG Maintaners. It provides the list of steps for how to release the crates and the
+considerations to make before releasing the create. It may provide use to consumers of the crate if/when we develop a
+release cadence.
+
+## Release cadence
+
+As of Feb 2024, there is no established cadence for the OpenTelemetry crates. The balance is required between too many
+breaking changes in a single release, and since we have instability flipping between implementations across 0.x
+releases. Perhaps once we've established a 1.0 we can consider a consistent release cadence.
+
+## Considerations
+
+A draft PR can be created, but before releasing consider the following:
+
+* Are there any pending pull requests which should be included in the next release?
+  * Are they blockers?
+* Are there any unresolved issues which should be resolved before the next release?
+* Bring it up at a SIG meeting, this can usually get some of these questions answered sooner than later. It will also
+  help establish a person to perform the release. Ideally this can be someone different each time to ensure that the
+  process is documented.
+
+## Steps to Release
+
+1. Create a release PR
+  - For each crate
+    - Bump appropriate version
+    - Update change logs to reflects
+    - Update API/SDK version as necessary
+  - If there's a large enough set of changes consider writing a migration guide.
+2. Merge the PR
+  - Get reviews from other Maintainers
+  - Ensure that there haven't been any interfering PRs
+3. Tag the release commit based on the [tagging convention](#tagging-convention)
+4. Create Github Release
+5. [Publish](#publishing-crates) to crates.io using the version as of the release commit
+6. Post to [#otel-rust](https://cloud-native.slack.com/archives/C03GDP0H023) on CNCF Slack.
+
+
+## Tagging Convention
+
+For each crate: it should be `<crate-name>-<version>` `<version>` being the simple `X.Y.Z`.
```

Comment:
should we consider aligning the same version for all crates? We moved all non-core crates to contrib, so should be easier if there is a single version for all core crates.?

Reply:
Pro: makes releasing simpler 
Con: Potentially creating more noise on the releases post 1.0

Not opposed in the short term. Perhaps we start by re-aligning them and we can revisit after 1.0 if it becomes more painful. I suppose after 1.x if nothing changes in a version we can just skip a version for that particular crate.

Reply:
Serial Version Pros
- Follows general convention 
- Avoids crate consumer confusion
- Removes dependence from api/sdk on exporter releases. A case where an exporter needs a breaking change without breaking API for example. 

Unified Version Pros
- Simplifies releasing for maintainers, maybe? 


Common Considerations 
- You still need to think about which crates to release. (This assumes we don't do worthless releases, which I'm not in favor of)

I could be swayed either way. Looking at it again I don't really see the advantage of the unified versioning. 

Am I missing any points? I can update my post.

Reply:
Cc: @djc

Reply:
Yeah, I wouldn't do unified versioning (instead documing a common reference point). Also, chances are it won't be a big deal after 1.0? Most crates will be at 1.0 together and will only evolve slowly from there.

Reply:
https://github.com/open-telemetry/opentelemetry-rust/tags We never used to push tags for each crate separately, so in some way, we were following the unified versioning (in terms of git tag only!)

Reply:
I realize that currently we only tag API SDK , but that's the part I was looking to change.

Reply:
From my perspective right now I'd like to document our current process, and we can approach the versioning question later. After the SIG we discussed this I believe since we didn't have a specific consensus we should continue with operating as we already do.